src/state/trajectory/logics.ts

```diff
@@ -547,6 +542,46 @@ const setConversionEngineLogic = createLogic({
     type: SET_CONVERSION_ENGINE,
 });
 
+const convertFileLogic = createLogic({
+    process(
+        deps: ReduxLogicDeps,
+        dispatch: <T extends AnyAction>(action: T) => T,
+        done
+    ) {
+        const { getState } = deps;
+
+        const { engineType, fileToConvert, fileName } =
+            getConversionProcessingData(getState());
+        const fileContents: Record<string, any> = {
+            fileContents: { fileContents: fileToConvert },
+            metaData: { trajectoryTitle: fileName },
+        };
+        dispatch(
+            clearSimulariumFile({
+                newFile: true,
+            })
+        );
+        const controller = getSimulariumController(getState());
+        // convert the file
+        dispatch(
+            setConversionStatus({
+                status: CONVERSION_ACTIVE,
+            })
+        );
+        controller
+            .convertAndLoadTrajectory(
+                netConnectionSettings,
+                fileContents,
+                engineType
+            )
+            .catch((err: Error) => {
+                console.error(err);
+            });
+        done();
+    },
+    type: CONVERT_FILE,
+});
+
```

Comment:
This is the logic that actually sends the file to Octopus for conversion. Because we are already at `/viewer` we are ready for Octopus to receive the converted file as soon as its ready.